multinode-demo/setup.sh

```diff
@@ -5,21 +5,33 @@ here=$(dirname "$0")
 source "$here"/common.sh
 
 usage () {
+  exitcode=0
+  if [[ -n "$1" ]]; then
+    exitcode=1
+    echo "Error: $*"
+  fi
   cat <<EOF
-usage: $0 [-n num_tokens] [-l] [-p]
+usage: $0 [-n num_tokens] [-l] [-p] [-t node_type]
 
 Creates a fullnode configuration
 
  -n num_tokens  - Number of tokens to create
  -l             - Detect network address from local machine configuration, which
                   may be a private IP address unaccessible on the Intenet (default)
  -p             - Detect public address using public Internet servers
+ -t node_type   - Create configuration files only for this kind of node.  Valid
+                  options are validator or leader.  Creates configuration files
+                  for both by default
+
 EOF
+  exit $exitcode
 }
 
 ip_address_arg=-l
 num_tokens=1000000000
-while getopts "h?n:lp" opt; do
+node_type_leader=true
```

Comment:
Using two booleans here seems clunky. Why not just `node_type_validator`?

Reply:
We have a 3rd state, "all", so down later in the file I was trying to avoid things like
```bash
if [[ $node_type = leader || $node_type = all ]]; then
```
where with this way I can just do

```
if $node_type_leader; then
```